lib/parse_date/int_from_string.rb

```diff
@@ -91,6 +63,63 @@ def self.year_int_valid?(year)
 
     protected
 
+    def earliest_year_bc_parsing(date_str)
```

Comment:
I am not in love with any of these method names;  suggestions for improvements welcome.  These are "good enough" I think.  

`parsing` vs `parse` ?  `parse_earliest_year` vs `earliest_year_parsing` ?

Reply:
What about dropping the `_parsing` portion? `earliest_year_bc` makes sense, I feel like including the "what it'd doing" in the name is what is throwing me off.

Reply:
Agree with @aaron-collier's sentiment here, and agree with @ndushay that it's good enough for now. My sense of the current state of the codebase is that—thanks to @ndushay's hard work and refactorings to date—we are getting **really** close to being able to do another (possibly large) refactoring to simplify the codebase by identifying one or more fruitful abstractions. I might save any renaming until that refactoring or immediately thereafter.